docs/tutorial-aks-demo.md

```diff
@@ -0,0 +1,386 @@
+# Secure AKS Secrets with Key Vault
+
+## Introduction
+
+A Secret in Kubernetes is meant to save sensitive and secure data like passwords, certificates, and keys. These Secret objects aren't secret or secure! They're encoded using Base64 and saved into *etcd*. Also, by default, any pod can access to them. As a solution, we can encrypt data at rest as explained in [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data). In this case, it's recommended to use a Key Management System (KMS) to securely save the encryption keys.
+
+KMS systems themselves have the feature to save the sensitive data like keys, secrets, files, and certificates. So, why not simply using it instead! Even more than just encryption, they offer powerful features like key rotation and expiration date.
+
+Now, to access KMS, a password is needed and should be saved in a Secret object in *etcd*! We returned back to the same problem. Fortunately, cloud providers have a solution. Instead of using a password to access KMS, the managed Kubernetes cluster will be granted access to KMS. In Azure, for example, AKS can connect to Key Vault or SQL Database… using an Identity. This connection could be done using the open-source [project Pod Identity](https://github.com/Azure/aad-pod-identity). It uses Azure AD to create an Identity and assign the roles and resources.
+
+Now that we have access to Key Vault, we can use its SDK or REST API in the application to retrieve the secrets. The SDK has support for .NET, Java, Python, JS, Ruby, PHP, etc. Or we can retrieve the secrets from a mounted volume. Historically, in Azure, this solution was implemented through [Kubernetes Key Vault Flex Volume](https://github.com/Azure/kubernetes-keyvault-flexvol). Now it's being deprecated. The new solution is [Azure Key Vault provider for Secret Store CSI driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure). Which is the Azure implementation of [Secrets Store CSI driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver).
+
+This tutorial will help you to securely retrieve secrets in Key Vault right from the Pod using Secrets Store CSI and AAD Pod Identity.
+
+## Setting up the environment
+
+To complete this workshop, we need az, kubectl and helm CLI. Also, we need to create the following resources in Azure:
+•	AKS cluster that uses Service Principal or Managed Identity through the variable $isAKSWithManagedIdentity.
+•	Key Vault with the following secrets: “DatabaseLogin: DbUserName” and “DatabasePassword: MyP@ssword123456”.
+•	Container Registry (ACR).
+We provision these resources from the Azure Portal or using the following Powershell script:
+
+```azurepowershell-interactive
+$suffix = "demo01"
+$subscriptionId = (az account show | ConvertFrom-Json).id
+$tenantId = (az account show | ConvertFrom-Json).tenantId
+$location = "westeurope"
+$resourceGroupName = "rg-" + $suffix
+$aksName = "aks-" + $suffix
+$aksVersion = "1.16.9"
```

Comment:
Can we consider using AKS default version?

Reply:
I thought it is better to use the latest supported version, but yes it keeps changing. 
Anyway, I changed it to 1.16.13 (default).